src/components/account/create.tsx

```diff
@@ -41,42 +54,144 @@ export class Container extends React.Component<Properties> {
     const {
       user: { nonce },
       currentAddress,
+    } = this.props;
+
+    if (nonce && currentAddress && prevProps.user?.nonce !== nonce) {
+      this.setState({
+        showDialog: true,
+      });
+    }
+
+    if (
+      this.state.showDialog &&
+      this.props.user.isLoading === false &&
+      prevProps.user &&
+      prevProps.user.data === null &&
+      this.props.user.data !== null
+    ) {
+      this.updateProfileImage();
+    }
+  };
+
+  shortAddress = (): string => {
+    const { currentAddress } = this.props;
+
+    if (!currentAddress) {
+      return '';
+    } else {
+      return [
+        currentAddress.slice(0, 6),
+        '...',
+        currentAddress.slice(-4),
+      ].join('');
+    }
+  };
+
+  closeDialog() {
+    this.setState({
+      showDialog: false,
+    });
+  }
+
+  updateProfileImage = async (): Promise<void> => {
+    const profileId = this.props.user.data.profileId;
+
+    try {
+      await this.props.updateImageProfile(profileId, this.state.profileImage);
+      this.closeDialog();
+    } catch (error) {
+      console.log('updating image profile failed. error:', error);
+    }
+  };
+
+  onSubmit = async (event?: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+
+    const {
+      user: { nonce },
       createAndAuthorize,
       fetchCurrentUser,
       inviteCode,
     } = this.props;
 
-    if (nonce && currentAddress && prevProps.user?.nonce !== nonce) {
-      const authResult = await createAndAuthorize(nonce, this.user(currentAddress), inviteCode)
-        .then((response) => response)
-        .catch((error) => {
-          const {
-            response: {
-              body: { code },
-            },
-          } = error;
-
-          if (/USER_HANDLE_ALREADY_EXISTS/.test(code)) {
-            console.log('Sorry that user handle has already been taken');
-          }
-        });
-
-      if (authResult) fetchCurrentUser();
+    const handle = this.shortAddress();
+
+    const user = { handle, firstName: this.state.displayName || handle, lastName: '' };
+
+    const authorizationResult = await createAndAuthorize(nonce, user, inviteCode)
+      .then((response) => response)
+      .catch((error) => {
+        const {
+          response: {
+            body: { code },
+          },
+        } = error;
+
+        if (/USER_HANDLE_ALREADY_EXISTS/.test(code)) {
+          console.log('Sorry that user handle has already been taken');
```

Comment:
I think we're going to need an error state on the `handle` field. A handle must be unique, the `createAndAuthorize` will fail if there's a duplicate handle.